core/src/main/java/jenkins/monitor/JavaVersionRecommendationAdminMonitor.java

```diff
@@ -78,8 +139,55 @@ public HttpResponse doAct(@QueryParameter String no) throws IOException {
             disable(true);
             return HttpResponses.forwardToPreviousPage();
         } else {
-            return new HttpRedirect("https://www.jenkins.io/redirect/upgrading-jenkins-java-version-8-to-11");
+            return new HttpRedirect("https://jenkins.io/redirect/java-support/");
         }
     }
 
+    @NonNull
+    private static LocalDate getEndOfLife() {
+        LocalDate endOfLife = SUPPORTED_JAVA_VERSIONS.get(Runtime.version().feature());
+        return endOfLife != null ? endOfLife : LocalDate.MAX;
+    }
+
+    @NonNull
+    private static Period getDeprecationPeriod() {
+        return Period.between(LocalDate.now(), getEndOfLife());
+    }
+
+    @NonNull
+    private static Severity getSeverity() {
+        return getDeprecationPeriod().getYears() < 1 ? Severity.DANGER : Severity.WARNING;
+    }
+
+    /**
+     * @return The current feature-release counter.
+     * @see Runtime#version()
+     */
+    @Restricted(DoNotUse.class)
+    public int getJavaVersion() {
+        return Runtime.version().feature();
+    }
+
+    /**
+     * @return The end of life date for the current Java version in the system default time zone.
+     */
+    @Restricted(DoNotUse.class)
+    public Date getEndOfLifeAsDate() {
+        return Date.from(getEndOfLife().atStartOfDay(ZoneId.systemDefault()).toInstant());
+    }
+
+    /**
+     * @return The severity of the administrative monitor, used to set the background color of the alert.
+     */
+    @Restricted(DoNotUse.class)
+    public String getSeverityAsString() {
+        return getSeverity().toString().toLowerCase(Locale.US);
+    }
+
+    private enum Severity {
+        SUCCESS,
+        INFO,
```

Comment:
Unused?

Reply:
Correct. Was there a point you were trying to make?

Reply:
Seems like it should be deleted. A random admin monitor doesn't need to be the reference for what kinds of CSS classes exist.

Reply:
Well this is a model object, so just like a Hibernate model object where every field has a getter and setter, some are going to be unused. But the model object should be complete so that if someone wants to use the model in a different way then they can do so in the future. Model objects are an exception to the general wisdom around deleting unused code, and (trust me) I love deleting code — the majority of my efforts as a programmer are around code cleanup rather than greenfielding. In any case, I've never added comments to unused getters in e.g. a Hibernate model object and I do not see a compelling reason to do so here.